plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/v1/GlueMetastoreModule.java

```diff
@@ -89,37 +101,49 @@ public RequestHandler2 createRequestHandler(OpenTelemetry openTelemetry)
                 .newRequestHandler();
     }
 
-    @Provides
-    @Singleton
-    @ForGlueHiveMetastore
-    public Executor createExecutor(GlueHiveMetastoreConfig hiveConfig)
+    private <C> void createExecutor(Class<? extends Annotation> annotationClass, String nameTemplate, Class<C> configClass, Function<C, Integer> threads)
     {
-        return createExecutor("hive-glue-partitions-%s", hiveConfig.getGetPartitionThreads());
+        install(conditionalModule(
+                configClass,
+                config -> threads.apply(config) > 1,
+                binder -> {
+                    binder.bind(Key.get(ExecutorService.class, annotationClass)).toInstance(newCachedThreadPool(daemonThreadsNamed(nameTemplate)));
+                    binder.bind(Key.get(BoundedExecutor.class, annotationClass)).toProvider(new BoundedExecutorProvider<>(annotationClass, configClass, threads)).in(Scopes.SINGLETON);
+                    binder.bind(Key.get(Executor.class, annotationClass)).to(Key.get(BoundedExecutor.class, annotationClass));
+                    closingBinder(binder).registerExecutor(Key.get(ExecutorService.class, annotationClass));
```

Comment:
We could maybe generalize this so that it is easier to Guice-create thread pools which will be closed later on.